test/base/builders/DAppOperationBuilder.sol

```diff
@@ -0,0 +1,114 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.21;
+
+import "forge-std/Test.sol";
+
+import { UserOperation } from "../../../src/contracts/types/UserCallTypes.sol";
+import { SolverOperation } from "../../../src/contracts/types/SolverCallTypes.sol";
+import { DAppOperation } from "../../../src/contracts/types/DAppApprovalTypes.sol";
+
+import { CallVerification } from "../../../src/contracts/libraries/CallVerification.sol";
+
+import { IDAppControl } from "../../../src/contracts/interfaces/IDAppControl.sol";
+import { IAtlasVerification } from "../../../src/contracts/interfaces/IAtlasVerification.sol";
+
+import "../../../src/contracts/types/DAppApprovalTypes.sol";
+
+contract DAppOperationBuilder is Test {
+    using CallVerification for UserOperation;
+
+    DAppOperation dappOperation;
+
+    function withFrom(address from) public returns (DAppOperationBuilder) {
+        dappOperation.from = from;
+        return this;
+    }
+
+    function withTo(address to) public returns (DAppOperationBuilder) {
+        dappOperation.to = to;
+        return this;
+    }
+
+    function withValue(uint256 value) public returns (DAppOperationBuilder) {
+        dappOperation.value = value;
+        return this;
+    }
+
+    function withGas(uint256 gas) public returns (DAppOperationBuilder) {
+        dappOperation.gas = gas;
+        return this;
+    }
+
+    function withMaxFeePerGas(uint256 maxFeePerGas) public returns (DAppOperationBuilder) {
+        dappOperation.maxFeePerGas = maxFeePerGas;
+        return this;
+    }
+
+    function withNonce(uint256 nonce) public returns (DAppOperationBuilder) {
+        dappOperation.nonce = nonce;
+        return this;
+    }
+
+    function withNonce(address atlasVerification, address account) public returns (DAppOperationBuilder) {
+        dappOperation.nonce = IAtlasVerification(atlasVerification).getNextNonce(account);
+        return this;
+    }
```

Comment:
IMO this should take `AtlasVerification` rather than `address` so we don't have to de-reference the address here.

This seems to lead to weird reverts that are hard to track down, so IMO we should try resolve everything in the test itself then pass instances rather than addresses when we can.

Reply:
I'm not sure I agree here, but don't have a strong opinion on it either. We could also expect an interface instead of an instance, we could then pass either an address or an `IAtlasVerification`, but not an `AtlasVerification`.
```solidity
function withNonce(IAtlasVerification atlasVerification, address account) public returns (DAppOperationBuilder) {
    dappOperation.nonce = atlasVerification.getNextNonce(account);
    return this;
}
```